zap/src/parser/convert.rs

```diff
@@ -207,6 +209,25 @@ impl<'src> Converter<'src> {
 		(value, span)
 	}
 
+	fn optional_str_opt(&mut self, name: &'static str, opts: &[SyntaxOpt<'src>]) -> (Option<String>, Option<Span>) {
+		let mut value = None;
+		let mut span = None;
+
+		for opt in opts.iter().filter(|opt| opt.name.name == name) {
+			if let SyntaxOptValueKind::Str(opt_value) = &opt.value.kind {
+				value = Some(self.str(opt_value).to_string());
+				span = Some(opt_value.span());
+			} else {
+				self.report(Report::AnalyzeInvalidOptValue {
+					span: opt.value.span(),
+					expected: "string",
+				});
+			}
+		}
+
+		(value, span)
+	}
+
```

Comment:
I'd rather not add `optional_str_opt` and instead have a `str_opt` with an empty default (`""`). Later we can check `types_output.is_empty()` to determine if we should output.

Reply:
Updated `types_output` to use `str_opt` instead.